web/src/typing_events.js

```diff
@@ -78,23 +77,16 @@ export function render_notifications_for_narrow() {
 }
 
 function get_key(event) {
-    let key;
-    let recipients;
-    switch (event.message_type) {
-        case "stream":
-            key = typing_data.get_topic_key(event.stream_id, event.topic);
-            break;
-        case "direct":
-            recipients = event.recipients.map((user) => user.user_id);
-            recipients.sort();
-
-            key = typing_data.get_direct_message_conversation_key(recipients);
-            break;
-        default:
-            blueslip.error("Unexpected message_type in typing event: " + event.message_type);
-            break;
+    if (event.message_type === "stream") {
+        return typing_data.get_topic_key(event.stream_id, event.topic);
     }
-    return key;
+    if (event.message_type === "direct") {
+        const recipients = event.recipients.map((user) => user.user_id);
+        recipients.sort();
+        return typing_data.get_direct_message_conversation_key(recipients);
+    }
+
+    return null;
```

Comment:
Where does returning `null` come from?

Reply:
OK I tweaked this to do a `throw new Error` here -- I think that's more correct as this should be an assertion about the set of valid types.

The problem with adding `null` here is that it means in order to understand the code, all the callers need to handle `null`; you've worked around that with type assertions, but those should be avoided when there's a better way.

Since we think this case is supposed to be impossible/the result of a buggy server, it's better to have the code say that explicitly.

Reply:
Got it thanks!!